application/src/main/java/org/togetherjava/tjbot/features/moderation/history/MessageListener.java

```diff
@@ -0,0 +1,46 @@
+package org.togetherjava.tjbot.features.moderation.history;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+import java.time.Instant;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+/**
+ * Listens for new message throughout the guild, then stores some metadata for each message in a
+ * database.
+ */
+public class MessageListener extends MessageReceiverAdapter {
```

Comment:
too generic name. adjust it to fit the purpose. this one is for storing messagemetadata. `MessageMetadataListener`.